src/test/java/com/redhat/exhort/integration/backend/BackendUtilsTest.java

```diff
@@ -72,90 +78,90 @@ void checkGeneratedWithNoRHDATokenAndShouldBeDifferentFromEachOther() {
-    Thread thread2 = new Thread(() -> reqId2.set(this.backendUtils.generateRequestId(randomUUID2)));
+    ;
+    Thread thread =
+        new Thread(() -> reqId.set(this.backendUtilsFakeTimer.generateRequestId(randomUUID)));
+    Thread thread2 =
+        new Thread(() -> reqId2.set(this.backendUtilsFakeTimer.generateRequestId(randomUUID2)));
     thread.start();
     thread2.start();
-
-    try {
-      thread.join();
-      thread2.join();
-    } catch (InterruptedException e) {
-      throw new RuntimeException(e);
-    } finally {
-      this.backendUtils.setClock(null);
-    }
+    thread.join();
+    thread2.join();
     assertFalse(reqId.get().equals(reqId2.get()));
   }
 
   @Test
-  void checkGeneratedWithSameTimeSameRHDATokenAndTheyShouldBeEquals() {
+  void checkGeneratedWithSameTimeSameRHDATokenAndTheyShouldBeEquals() throws InterruptedException {
 
     String randomUUID = UUID.randomUUID().toString();
 
     AtomicReference<String> reqId = new AtomicReference();
     AtomicReference<String> reqId2 = new AtomicReference();
-    LocalDate fakeDate = LocalDate.of(2024, 1, 1);
-    Clock clock =
-        Clock.fixed(fakeDate.atStartOfDay().toInstant(ZoneOffset.UTC), ZoneId.systemDefault());
-    this.backendUtils.setClock(clock);
-    Thread thread = new Thread(() -> reqId.set(this.backendUtils.generateRequestId(randomUUID)));
-    Thread thread2 = new Thread(() -> reqId2.set(this.backendUtils.generateRequestId(randomUUID)));
+    Thread thread =
+        new Thread(() -> reqId.set(this.backendUtilsFakeTimer.generateRequestId(randomUUID)));
+    Thread thread2 =
+        new Thread(() -> reqId2.set(this.backendUtilsFakeTimer.generateRequestId(randomUUID)));
     thread.start();
     thread2.start();
-    try {
-      thread.join();
-      thread2.join();
-    } catch (InterruptedException e) {
-      throw new RuntimeException(e);
-    } finally {
-      this.backendUtils.setClock(null);
-    }
-
+    thread.join();
+    thread2.join();
     assertTrue(reqId.get().equals(reqId2.get()));
   }
 
   @Test
-  void checkGeneratedWithSameTimeWithNullRHDATokenAndTheyShouldBeDifferent() {
+  void checkGeneratedWithSameTimeWithNullRHDATokenAndTheyShouldBeDifferent()
+      throws InterruptedException {
 
     String randomUUID = UUID.randomUUID().toString();
 
     AtomicReference<String> reqId = new AtomicReference();
     AtomicReference<String> reqId2 = new AtomicReference();
-    LocalDate fakeDate = LocalDate.of(2024, 1, 1);
-    Clock clock =
-        Clock.fixed(fakeDate.atStartOfDay().toInstant(ZoneOffset.UTC), ZoneId.systemDefault());
-    this.backendUtils.setClock(clock);
-    Thread thread = new Thread(() -> reqId.set(this.backendUtils.generateRequestId(null)));
-    Thread thread2 = new Thread(() -> reqId2.set(this.backendUtils.generateRequestId(null)));
+    Thread thread = new Thread(() -> reqId.set(this.backendUtilsFakeTimer.generateRequestId(null)));
```

Comment:
why not use the same concurrency approach for all the tests? Maybe you can create a helper lambda where the only difference is the constructor

Reply:
> why not use the same concurrency approach for all the tests? Maybe you can create a helper lambda where the only difference is the constructor

@ruromero Do you mean use Executor with threads pool for 2 threads? i think it's a bit overkill :) i can refactor the generation of thread with an helper method that i'll pass it the proper parameters to invoke the tested method

Reply:
@ruromero i extracted method with parameters to create various threads that will do the work.